### PR TITLE
Added listener on menueditor extension, so we can set the needed data.

### PR DESCRIPTION
**Pull request by SantinoPetrovic**

### CODE REVIEW

**Comment by SantinoPetrovic** [comment]
Oh sorry about that... It doesn't support type yet... Text is default.

File: src/TranslateExtension.php [hunk 1]
```diff
@@ -384,4 +402,17 @@ protected function getDefaultConfig()
...
+
+    /**
+     * Set Fields for field builder event.
+     */
+    public function onRequest(FieldBuilderEvent $event)
+    {
+        $labels = [];
+        $config = $this->getConfig();
+        foreach ($config['locales'] as $language) {
+            $labels[] = array($language['slug'].'label' => $language['slug'].'label');
```
Comment (SahAssar) [inline src/TranslateExtension.php]: Just a few small things here: Could you change to the short array syntax and put some spacing between the concat operators? something like this: `[$language['slug'] . 'label' => $language['slug'] . 'label'];`

File: src/TranslateExtension.php [hunk 2]
```diff
@@ -187,7 +194,18 @@ function ($app) {
...
+        );        
+        $app['translate.locales'] = $app->share(
+            function ($app) {
+                $config = $app['translate.config'];
+                if (count($config->getLocales()) === 0) {
+                    return;
+                } else {
+                   return $config->getLocales();
+                }
+
```
Comment (SahAssar) [inline src/TranslateExtension.php]: I think this line can be removed, no?

File: src/TranslateExtension.php [hunk 3]
```diff
@@ -187,7 +194,18 @@ function ($app) {
 
                 return $defaultSlug;
             }
-        );
+        );        
+        $app['translate.locales'] = $app->share(
```
Comment (SahAssar) [inline src/TranslateExtension.php]: A blankline above here before starting the new block would be nice :)

File: src/TranslateExtension.php [hunk 4]
```diff
@@ -6,6 +6,7 @@
 use Bolt\Asset\Target;
 use Bolt\Controller\Zone;
 use Bolt\Extension\Animal\Translate\Config;
+use Bolt\Extension\Bacboslab\Menueditor\Event\FieldBuilderEvent;
```
Comment (SahAssar) [inline src/TranslateExtension.php]: This class (FieldBuilderEvent) does not seem to be in this PR or in the current repo, could you add it?

File: src/TranslateExtension.php (hunk 3 above)
Comment (SantinoPetrovic) [inline src/TranslateExtension.php]: Between line 197 and 198?

Comment (SahAssar) [inline src/TranslateExtension.php]: Yep

**Comment by SantinoPetrovic** [comment]
I believe that's it. Take a new look :)

**Comment by SantinoPetrovic** [comment]
Just a reminder, don't forget this PR :)

**Comment by SahAssar** [comment]
Sorry for the delay, commented in https://github.com/bacbos/bolt-menu-editor/pull/67#issuecomment-345360290

**Comment by SahAssar** [comment]
Closing cause the mentioned issues have not been resolved.

**Pull request closed without merging**